Kernel/Arch/riscv64/PCI/Initializer.cpp

```diff
@@ -147,8 +151,48 @@ void initialize()
         case ControllerCompatible::Unknown:
             VERIFY_NOT_REACHED(); // This should have been rejected earlier
         }
+
+        auto maybe_ranges = node.get_property("ranges"sv);
+        if (maybe_ranges.has_value()) {
+            auto address_cells = node.get_property("#address-cells"sv).value().as<u32>();
+            VERIFY(address_cells > 1); // Additional cell for PCI address metadata
```

Comment:
You could also `VERIFY(address_cells == 3)` here, as the bindings require #address-cells == 3: https://github.com/devicetree-org/dt-schema/blob/main/dtschema/schemas/pci/pci-bus-common.yaml#L70
If the PCI controller node doesn't have an #address-cells of 3, we likely can't rely on the child-bus-address being a valid OpenFirmware PCI address.

Reply:
done